pkg/volume/csi/csi_mounter.go

```diff
@@ -360,13 +361,17 @@ func (c *csiMountMgr) TearDownAt(dir string) error {
 	ctx, cancel := context.WithTimeout(context.Background(), csiTimeout)
 	defer cancel()
 
+	var retErrors []error
 	if err := csi.NodeUnpublishVolume(ctx, volID, dir); err != nil {
-		return errors.New(log("mounter.TearDownAt failed: %v", err))
+		retErrors = append(retErrors, errors.New(log("mounter.TearDownAt failed: %v", err)))
```

Comment:
If we remove the mount dir, what ensures NodeUnpublishVolume gets called in the future?

Our contract with CSI is that NodeUnpublishVolume must succeed before we allow pod teardown.  If a CSI plugin blocks teardown, that's a bug in the CSI plugin.

Reply:
Looking at a few callers, I see the following pattern (e.g. downwardAPIVolumeMounter#SetUpAt):
```
	defer func() {
		// Clean up directories if setup fails
		if !setupSuccess {
...
			tearDownErr := unmounter.TearDown()
```

Reply:
In Setup:
```
	// create target_dir before call to NodePublish
	if err := os.MkdirAll(dir, 0750); err != nil && !corruptedDir {
...
	err = csi.NodePublishVolume(
		ctx,
		volumeHandle,
```

Reply:
ephemeral volumes don't have same problems as CSI volumes. I agree that we should not delete the mount directory if `NodeUnpublish` fails.  Among other things - a volume which has its `unmount` operation as failing, is still considered in-use by volumemanager and another pod *could* attempt to use it.